src/common/common.h

```diff
@@ -169,14 +169,15 @@ class GPUSet {
     int n =  AllVisibleImpl::AllVisible();
     return Range(0, n);
   }
-  /*! \brief Ensure gpu_id is correct, so not dependent upon user knowing details */
-  static int GetDeviceIdx(int gpu_id) {
-    auto devices = AllVisible();
-    CHECK(!devices.IsEmpty()) << "Empty device.";
-    return (std::abs(gpu_id) + 0) % devices.Size();
-  }
+
   /*! \brief Counting from gpu_id */
   GPUSet Normalised(int gpu_id) const {
+    int n_devices_visible = AllVisible().Size();
+    CHECK(Size() + gpu_id <= n_devices_visible)
+        << "n_gpu + gpu_id should less than or equal to total number of available devices."
+        << "\n n_gpu: " << Size()
+        << ", gpu_id: " << gpu_id
+        << ", number of available devices: " << n_devices_visible;
```

Comment:
Lets remove Unnormalised(), I think it is wrong as the range can be larger than the number of GPUs. This unnormalised/normalised terminology is confusing and I can see it somehow has different meanings across different parts of the code. From now on whenever a device index is referred to it means the physical device ordinal of the GPU.